Tests/Engine/DataProviders/ApiDataProviderTests.cs

```diff
@@ -51,5 +43,20 @@ public void OutOfDateTest(Resolution resolution, int days, bool expected)
             Assert.AreEqual(expected, test);
             File.Delete(path);
         }
+
+        [TestCase("forex/oanda/minute/eurusd/20131011_quote.zip")]
+        [TestCase("equity/usa/factor_files/tsla.csv")]
+        [TestCase("equity/usa/factor_files/factor_files_20210601.zip")]
+        public void DownloadFiles(string file)
+        {
+            var dataProvider = new ApiDataProvider();
+            var path = Path.Combine(Globals.DataFolder, file);
+            var stream = dataProvider.Fetch(path);
+
+            Assert.IsNotNull(stream);
+            stream.Dispose();
+            
+            Assert.IsTrue(File.Exists(path));
+        }
```

Comment:
We should add some tests for map files, also the NeedUpdate logic checking how old files are and assert it doesn't download twice